book/appendix/wallaroo-command-line-options.md

```diff
@@ -4,66 +4,100 @@ Every Wallaroo option exposes a set of command-line options that are used to con
+  --phone-home/-p [Sets address for phone home connection]
+  --external/-e [Sets address for external message channel]
+  --worker-count/-w +[Sets cluster initialier's total number of workers,
+    including cluster initializer itself]
+  --name/-n +[Sets name for this worker]
+
+  --metrics/-m [Sets address for external metrics (e.g. monitoring hub)]
+  --cluster-initializer/-t [Sets this process as the cluster
+    initializing process (that status is meaningless after init is done)]
+  --resilience-dir/-r [Sets directory to write resilience files to,
+    e.g. -r /tmp/data (no trailing slash)]
+  --log-rotation [Enables log rotation. Default: off]
+  --event-log-file-size/-l [Optionally set a file size for triggering
+    event log file rotation. If no file size is set, log rotation is only
+    triggered by external control messages sent to the address used with
+    --external]
+
+  --join/j [When a new worker is joining a running cluster, pass the
+    control channel address of any worker as the value for this
+    parameter]
+  --stop-world/u [Sets pause before state migration after the stop the
+    world]
+
+-------------------------------------------------------------------------
+To set Wallaroo modes, compile with the following -D flags:
+  resilience - Sets resilience mode
+  autoscale - Sets autoscaling mode
+  clustering - Sets clustering mode
 ```
 
 Wallaroo currently supports one input stream per pipeline. We provide comma-separated IP addresses for TCP source listeners via the `--in` parameter. Currently, the order of these addresses corresponds to the order in which pipelines are defined in your application code.
 
-Wallaroo currently supports one sink per app. Use the `--out` parameter to specify the address that this sink will write out to over TCP.
+Wallaroo currently supports one sink per application. If you are using a `TCPSInk`, use the `--out` parameter to specify the address (in a `host:port` format) that this sink will write out to over TCP. For other types of sinks, please refer to the specific sink's documentation.
+
+In order to monitor metrics, we must specify the target address for metrics data via the `--metrics/-m` parameter.
+
+## Machida specific parameters
+
+In addition to the Wallaroo command line paramters, Machida, the python-wallaroo interface, takes the additional argument
+
+```bash
+  --application-module [Specify the Machida application module]
+```
 
-In order to monitor metrics, we must specify the target address for metrics data via the `--metrics` parameter.
+`--application-module` specifies the _name_ that machida will attempt to import as the Python Wallaroo application file. For example, if you write a Python Wallaroo application and save it as `my_application.py`, then you should provide that name to machida as `--application-module my_application`.
+The application should be available to import from the location where `machida` is run. You can achieve this by:
```

Comment:
this is nice and all but this is not about the command line anymore. i think this shouldn't be here.

Reply:
I think this is relevant information to the command line option `--application-module`, as it pertains to limitations in terms of what it can take. For example, it isn't obvious that you can't use `--application-module <absolute path to my_application.py>` if this isn't included.

Reply:
Or did you only mean that the "ways to achieve this" part could be left out?

Reply:
I agree with @SeanTAllen, this is out of scope for the purposes of this document.

Reply:
I'm still not clear if this is referring to the PYTHONPATH setting, or the entire section?

Reply:
@nisanharamati sorry about the confusion. The `PYTHONPATH` section feels out of scope, but the part about how to use the `--application-module` should stay.

To be clear, just keep:
> `--application-module` specifies the _name_ that machida will attempt to import as the Python Wallaroo application file. For example, if you write a Python Wallaroo application and save it as `my_application.py`, then you should provide that name to machida as `--application-module my_application`.